model/adb_auth.go

```diff
@@ -122,16 +204,60 @@ func getUsers(db *sqlx.DB, options GetUserOptions) ([]ADBUser, error) {
 	return users, nil
 }
 
+func getUsersRoles(db *sqlx.DB, options GetUsersRolesOptions) ([]UserRole, error) {
+	query := selectUsersRolesBaseQuery
+	/*
+	  var queryArgs []interface{}
+	  var whereClause []string
+
+	  if len(options.Users) != 0 {
+	    var userIds = []int{4, 6, 7}
+	    //for _, user := range options.Users {
+	    //  userIds = append(userIds, strconv.Itoa(user.ID))
+	    //}
+	    whereClause = append(whereClause, "ur.user_id IN (?)")
+	    queryArgs = append(queryArgs, userIds)
+	  }
+
+	  if len(options.Roles) != 0 {
+	    whereClause = append(whereClause, "ur.role IN (?)")
+	    queryArgs = append(queryArgs, options.Roles)
+	  }
+
+	  if len(whereClause) != 0 {
+	    query += ` WHERE ` + strings.Join(whereClause, " AND ")
+	  }
+	*/
+	fmt.Println(query)
+	var userRoles []UserRole
+	err := db.Select(&userRoles, query)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to select UserRoles")
+	}
+
+	if len(userRoles) == 0 {
+		return nil, nil
+	}
+
+	return userRoles, nil
+}
+
 func buildUserJSONArray(users []ADBUser) []UserJSON {
 	var usersJSON []UserJSON
 
 	for _, u := range users {
+		var roles []string
+		for _, userRole := range u.Roles {
+			roles = append(roles, userRole.Role)
+		}
 
 		usersJSON = append(usersJSON, UserJSON{
 			ID:       u.ID,
 			Email:    u.Email,
 			Admin:    u.Admin,
 			Disabled: u.Disabled,
+			Roles:    roles,
```

Comment:
Can't you just use `u.Roles` here?

Reply:
I wanted to avoid sending back a complex data structure in the Roles field for the user's JSON. `u.Roles` would be an array of objects like `{UserID: 1, Role: "admin"}`. We don't need the `UserID`.

Here I'm just extracting the Roles into a simple array of strings: `["admin"]`.

Reply:
Ohh, I missed u.Roles wasn't `[]string`. Sorry about that.